server_src/constants/state.constants.js

```diff
@@ -0,0 +1,222 @@
+  Paused: "Paused",
+  Resuming: "Resuming",
+  Finishing: "Finishing",
+  Error: "Error",
+  OfflineAfterError: "Offline after error",
+  UnknownState: "Unknown State ()" // Unknown State (...) needs proper parsing
+};
+
+// All states of the app. Nice to share between server and client
+const PSTATE = {
+  Offline: "Offline",
+  GlobalAPIKey: "Global API Key Issue",
+  Searching: "Searching...",
+  "Error!": "Error!",
+  "No-API": "No-API",
+  Disconnected: "Disconnected",
+  Starting: "Starting",
+  Operational: "Operational",
+  Paused: "Paused",
+  Printing: "Printing",
+  Pausing: "Pausing",
+  Cancelling: "Cancelling",
+  "Offline after error": "Offline after error",
+  Complete: "Complete",
+  Shutdown: "Shutdown",
+  Online: "Online"
+};
+
+const mapStateToColor = (state) => {
+  if (state === PSTATE.Loading) {
+    return { name: "dark", hex: "#262626", category: CATEGORY.Idle };
+  }
+  if (state === PSTATE.Operational) {
+    return { name: "dark", hex: "#262626", category: CATEGORY.Idle };
+  }
+  if (state === PSTATE.Paused) {
+    return { name: "warning", hex: "#583c0e", category: CATEGORY.Idle };
+  }
+  if (state === PSTATE.Printing) {
+    return { name: "warning", hex: "#583c0e", category: CATEGORY.Active };
+  }
+  if (state === PSTATE.Pausing) {
+    return { name: "warning", hex: "#583c0e", category: CATEGORY.Active };
+  }
+  if (state === PSTATE.Cancelling) {
+    return { name: "warning", hex: "#583c0e", category: CATEGORY.Active };
+  }
+  if (state === PSTATE.Starting) {
+    return { name: "warning", hex: "#583c0e", category: CATEGORY.Active };
+  }
+  if (state === PSTATE.GlobalAPIKey) {
+    return { name: "danger", hex: "#2e0905", category: CATEGORY["Error!"] };
+  }
+  if (state === PSTATE["Error!"]) {
+    return { name: "danger", hex: "#2e0905", category: CATEGORY["Error!"] };
+  }
+  if (state === PSTATE.Offline) {
+    return { name: "danger", hex: "#2e0905", category: CATEGORY.Offline };
+  }
+  if (state === PSTATE["Searching"]) {
+    return { name: "danger", hex: "#2e0905", category: CATEGORY.Offline };
+  }
+  if (state === PSTATE.Disconnected) {
+    return { name: "danger", hex: "#2e0905", category: CATEGORY.Disconnected };
+  }
+  if (state === PSTATE["No-API"]) {
+    return { name: "danger", hex: "#2e0905", category: CATEGORY.Offline };
+  }
+  if (state === PSTATE.Complete) {
+    return { name: "success", hex: "#00330e", category: CATEGORY.Complete };
+  }
+  if (state === PSTATE.Shutdown) {
+    return { name: "danger", hex: "#2e0905", category: CATEGORY.Offline };
+  }
+  if (state === PSTATE.Online) {
+    return { name: "success", hex: "#00330e", category: CATEGORY.Idle };
+  }
+  if (state === PSTATE["Offline after error"]) {
+    return { name: "danger", hex: "#2e0905", category: CATEGORY["Error!"] };
+  }
+
+  // TODO is this a smart idea? No error?
+  console.warn("PSTATE not recognized:", state);
+  return { name: "warning", hex: "#583c0e", category: CATEGORY.Active };
+};
```

Comment:
A missed state in my opinion shouldn't stop the printer connection from working to me. I defaulted back to active because it was generally there we're I may have missed a state.

Reply:
I guess we should log this in dev mode only to spot OP API changes we didnt foresee.

Reply:
Yeah that sounds like a good plan. Seeing the OP file now it shouldn't be too hard to keep up with them anyway. I just think a fall back to offline is a bit confusing to the user. How about we fall back to "unknown" state? Could be ignored then in the rest of the backend / client as nothing is looking for it

Reply:
Your right Active is also confusing btw aha

Reply:
Nice I will make sure to pick this up in le big one